fearless/Modules/Staking/Model/StakingSharedState.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+final class StakingSharedState {
+    let settings: StakingAssetSettings
+    private(set) var eraValidatorService: EraValidatorServiceProtocol
+    private(set) var rewardCalculationService: RewardCalculatorServiceProtocol
+    let stakingLocalSubscriptionFactory: StakingLocalSubscriptionFactoryProtocol
+    let stakingAnalyticsLocalSubscriptionFactory: StakingAnalyticsLocalSubscriptionFactoryProtocol
+
+    init(
+        settings: StakingAssetSettings,
+        eraValidatorService: EraValidatorServiceProtocol,
+        rewardCalculationService: RewardCalculatorServiceProtocol,
+        stakingLocalSubscriptionFactory: StakingLocalSubscriptionFactoryProtocol,
+        stakingAnalyticsLocalSubscriptionFactory: StakingAnalyticsLocalSubscriptionFactoryProtocol
+    ) {
+        self.settings = settings
+        self.eraValidatorService = eraValidatorService
+        self.rewardCalculationService = rewardCalculationService
+        self.stakingLocalSubscriptionFactory = stakingLocalSubscriptionFactory
+        self.stakingAnalyticsLocalSubscriptionFactory = stakingAnalyticsLocalSubscriptionFactory
+    }
+
+    func replaceEraValidatorService(_ newService: EraValidatorServiceProtocol) {
+        eraValidatorService = newService
+    }
+
+    func replaceRewardCalculatorService(_ newService: RewardCalculatorServiceProtocol) {
+        rewardCalculationService = newService
+    }
```

Comment:
I would make it more readable by changing to 
```swift
func replaceEraValidatorService(with newService: EraValidatorServiceProtocol)
func replaceRewardCalculatorService(with newService: RewardCalculatorServiceProtocol)
```
These functions look like syntactic sugar. Are they?